Common/Securities/Option/OptionStrategy.cs

```diff
@@ -74,35 +98,66 @@ public class OptionLegData
             public decimal Strike { get; set; }
 
             /// <summary>
-            /// Type of order that is to be sent to the market on strategy execution
+            /// Creates a new instance of <see cref="OptionLegData"/> from the specified parameters
             /// </summary>
-            public OrderType OrderType { get; set; }
+            public static OptionLegData Create(int quantity, Symbol symbol, OrderType orderType = OrderType.Market, decimal? orderPrice = null)
+            {
+                return new OptionLegData
+                {
+                    Symbol = symbol,
+                    Quantity = quantity,
+                    OrderType = orderType,
+                    Expiration = symbol.ID.Date,
+                    OrderPrice = orderPrice ?? 0m,
+                    Right = symbol.ID.OptionRight,
+                    Strike = symbol.ID.StrikePrice
+                };
+            }
 
             /// <summary>
-            /// Order limit price of the leg in case limit order is sent to the market on strategy execution
+            /// Invokes the <paramref name="optionHandler"/>
             /// </summary>
-            public decimal OrderPrice { get; set; }
+            public override void Invoke(Action<UnderlyingLegData> underlyingHandler, Action<OptionLegData> optionHandler)
+            {
+                optionHandler(this);
+            }
         }
 
         /// <summary>
         /// This class is a POCO containing basic data for the underlying leg of the strategy
         /// </summary>
-        public class UnderlyingLegData
+        public class UnderlyingLegData : LegData
         {
             /// <summary>
-            /// Quantity multiplier used to specify proper scale (and direction) of the leg within the strategy
+            /// Creates a new instance of <see cref="UnderlyingLegData"/> for the specified <paramref name="quantity"/> of underlying shares.
             /// </summary>
-            public int Quantity { get; set; }
+            public static UnderlyingLegData Create(int quantity, Symbol symbol, OrderType orderType = OrderType.Market, decimal? orderPrice = null)
+            {
+                var data = Create(quantity, orderType, orderPrice);
+                data.Symbol = symbol;
+                return data;
+            }
 
             /// <summary>
-            /// Type of order that is to be sent to the market on strategy execution
+            /// Creates a new instance of <see cref="UnderlyingLegData"/> for the specified <paramref name="quantity"/> of underlying shares.
             /// </summary>
-            public OrderType OrderType { get; set; }
+            public static UnderlyingLegData Create(int quantity, OrderType orderType = OrderType.Market, decimal? orderPrice = null)
```

Comment:
I think the issue is in creating the correct subtype -- if there was only one `Create` method it would require an additional parameter specifying whether we want underlying or option. In such situations, I think it's an anti-pattern to collapse into one method using an argument to switch. Two separate, yet similar, static methods are more explicit and not that it's important, but also gives the compiler a bit more info for optimization.